cclient/iota_client_extended_api.c

```diff
@@ -249,3 +249,80 @@ retcode_t iota_client_get_account_data(iota_client_service_t const* const serv,
   get_balances_res_free(&balances_res);
   return ret_code;
 }
+
+retcode_t iota_client_get_bundle(iota_client_service_t const* const serv,
+                                 flex_trit_t const* const tail_hash,
+                                 transaction_objs_t* out_tx_objs) {
+  retcode_t ret_code = RC_OK;
+
+  log_debug(CCLIENT_EXTENDED_LOGGER_ID, "[%s:%d]\n", __func__, __LINE__);
+
+  return iota_client_traverse_bundle(serv, tail_hash, out_tx_objs);
+}
+
+retcode_t iota_client_traverse_bundle(iota_client_service_t const* const serv,
+                                      flex_trit_t const* const trunk_tx,
+                                      transaction_objs_t *const tx_objs) {
+  retcode_t ret_code = RC_OK;
+  get_trytes_req_t* get_trytes_req = NULL;
+  get_trytes_res_t* get_trytes_res = NULL;
+  iota_transaction_t tx;
+  flex_trit_t* tmp_trytes;
+  flex_trit_t* trunk_hash = (flex_trit_t *)trunk_tx;
+  
+  log_debug(CCLIENT_EXTENDED_LOGGER_ID, "[%s:%d]\n", __func__, __LINE__);
+
+  do {
+    // Create request and response to get_trytes API
+    get_trytes_req = get_trytes_req_new();
```

Comment:
We should reuse these instead of reallocing?

Reply:
Ah, yes